tests/amsterdam/eip7805_forkchoice_enforced_inclusion_lists/test_cases.md

```diff
@@ -0,0 +1,21 @@
+| `test_focil_builder_ignores_invalid_inclusion_list_transactions` | Ensure `produce_execution_payload` does not include `inclusionListTransactions` that are invalid against the parent state. | The EL is asked to build a block with an inclusion list containing a mix of valid and invalid (e.g., bad nonce) transactions. | The produced block MUST include the valid `inclusionListTransactions` and MUST NOT include the invalid ones. | 🟡 Planned |
+| `test_focil_block_builder_handles_empty_inclusion_list` | Ensure a valid block is built with zero `inclusionListTransactions`. | The EL is asked to build a block with zero `inclusionListTransactions`. | The builder MUST produce a valid block. | 🟡 Planned |
+| `test_focil_builder_handles_inclusion_list_tx_becoming_invalid` | Ensure builder correctly omits an inclusion list transaction that becomes invalid due to a preceding inclusion list transaction. | The `inclusionListTransactions` are `[tx_A, tx_B]`. Both are from the same EOA with nonce `N`. Both are valid against the parent state. | The builder MUST include `tx_A`. `tx_B` (with nonce `N`) becomes invalid and MUST be omitted from the block. | 🟡 Planned |
+| `test_focil_builder_handles_same_sender_insufficient_funds_after_first_tx` | Ensure builder correctly omits a transaction that becomes unaffordable after a preceding transaction from the same sender executes. | The `inclusionListTransactions` are `[tx_A, tx_B]`. Both are from the same EOA with sequential nonces (0, 1). Both valid against parent state. `tx_A` transfers most funds away, leaving insufficient balance for `tx_B`'s gas. | The builder MUST include `tx_A`. `tx_B` becomes unaffordable after `tx_A` and MUST be omitted from the block. | 🟡 Planned |
+| `test_focil_builder_handles_full_block_with_partial_inclusion_list` | Ensure builder correctly handles when block gas limit is reached and not all `inclusionListTransactions` can fit. | The `inclusionListTransactions` contain valid independent transactions whose total gas exceeds the block gas limit. | The builder MUST include as many transactions from `inclusionListTransactions` as fit within the block gas limit. Remaining transactions are omitted (block full). | 🟡 Planned |
+| `test_focil_builder_handles_same_sender_wrong_nonce_order` | Ensure builder correctly handles `inclusionListTransactions` with transactions from the same EOA in wrong nonce order. | The `inclusionListTransactions` are `[tx_B, tx_A]` where `tx_A` has nonce 0 and `tx_B` has nonce 1. Both are from the same EOA. When processed in order, `tx_B` is invalid (nonce too high). | The builder MUST include `tx_A` first, and then include `tx_B` rechecking its validity. | 🟡 Planned |
```

Comment:
While the test itself is valid, IL transactions are not ordered in any sense. It's just a set. An example of bad nonce error would be `[tx_A, tx_B]` where both have same nonce.

Reply:
> the builder impl is left up to client

Speaking of which, not sure if this is possible, but maybe we could test like this flow:
1) IL <= getInclusionList()
2) newPayload should be fine